### PR TITLE
Added TCP and WebSocket connection proxy configuration

### PR DESCRIPTION
**Pull request by OleksiiButeskul**

### CODE REVIEW

File: steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.avenga.steamclient.model.proxy;
+
+import lombok.Getter;
+
+import java.net.Proxy;
+
+@Getter
+public class ProxyState {
```
Comment (andyislegend) [inline steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java:8]: its not ThreadSafe. Should it?

File: steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    private Proxy proxy;
+    private int connectionFailureCount;
```
Comment (andyislegend) [inline steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java:10]: Atomic?

File: steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java (hunk 1 above)
Comment (OleksiiButeskul) [reply steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java:8]: One Steam client maintain only one connection. These objects are maintained by internal logic, which user can't impact directly. Even if we will use `SteamClient ` in multi-thread logic, rule of the first statement will be true. I will change `connectionFailureCount` to Atomic to be on safe side.

File: steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class ProxyState {
+    private Proxy proxy;
+    private AtomicInteger connectionFailureCount;
+
+    public ProxyState(Proxy proxy) {
+        this.proxy = proxy;
+    }
+
+    public void incrementFailureCount() {
+        connectionFailureCount.incrementAndGet();
```
Comment (andyislegend) [inline steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java:18]: I would prefare ::compareAndSet()

Comment (OleksiiButeskul) [reply steam-client/src/main/java/com/avenga/steamclient/model/proxy/ProxyState.java]: Done

**Pull request merged**